trieste/bayesian_optimizer.py

```diff
@@ -22,15 +22,22 @@
 import traceback
 from collections.abc import Mapping
 from dataclasses import dataclass
-from typing import Dict, Generic, TypeVar, cast, overload
+from typing import Dict, Generic, Optional, TypeVar, cast, overload
 
+import absl
 import numpy as np
 import tensorflow as tf
-from absl import logging
 
+try:
+    import pandas as pd
+    import seaborn as sns
```

Comment:
so we are introducing dependencies for this?

Reply:
Yes. And if you enable any summaries that require seaborn but don't have it installed you get a warning message telling you this.

Reply:
is it possible to have these in setup.py as optional?

Reply:
what I want to say, its hard to see this dependency unless you run into this error

Reply:
Good suggestion. I already knew poetry supported this, but looks like setup.py does too via something like this. I'll add it and check it works.

```python
    extras_require = { "plotting": ["seaborn"] }
```